homeassistant/components/sensor/version.py

```diff
@@ -1,54 +1,108 @@
 """
-Support for displaying the current version of Home Assistant.
+Sensor that can display the current Home Assistant versions.
 
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/sensor.version/
 """
 import logging
+from datetime import timedelta
 
 import voluptuous as vol
 
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 from homeassistant.components.sensor import PLATFORM_SCHEMA
-from homeassistant.const import __version__, CONF_NAME
+from homeassistant.const import CONF_NAME, CONF_SOURCE
 from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['pyhaversion==2.0.1']
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_BETA = 'beta'
+CONF_IMAGE = 'image'
+
+DEFAULT_IMAGE = 'default'
 DEFAULT_NAME = "Current Version"
+DEFAULT_SOURCE = 'local'
+
+ICON = 'mdi:package-up'
+
+TIME_BETWEEN_UPDATES = timedelta(minutes=5)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_BETA, default=False): cv.boolean,
+    vol.Optional(CONF_IMAGE, default=DEFAULT_IMAGE): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_SOURCE, default=DEFAULT_SOURCE): cv.string,
 })
 
 
 async def async_setup_platform(
         hass, config, async_add_entities, discovery_info=None):
     """Set up the Version sensor platform."""
+    from pyhaversion import Version
+    beta = config.get(CONF_BETA)
+    image = config.get(CONF_IMAGE).lower()
```

Comment:
Put all config modifications in the config schema. Use `vol.Lower`.

Reply:
Better now? :)